### PR TITLE
New Service and Validation

### PR DESCRIPTION
**Pull request by ZHIFA-CHIU**
Adding a new service to re-use some of the code, adding validation features of add / update part methods

### CODE REVIEW

File: src/main/java/com/mercuryCyclists/Inventory/service/ProductService.java [hunk 1]
```diff
@@ -93,15 +93,27 @@ public boolean addPart(Long productId, Part part) {
...
-    public Part updatePart(Long id, Part part) {
-        Part p = partRepository.findById(id).orElse(null);
-        if (p == null) return null;
+    public Part updatePart(Long productId, Long partId, Part part) {
+        Product product = productRepository.findById(productId).orElse(null);
+        Part p = partRepository.findById(partId).orElse(null);
+        if (p == null || product == null || !updatePartHelper(product, p)) return null;
         p.setName(part.getName());
         p.setDescription(part.getDescription());
         return partRepository.save(p);
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Inventory/service/ProductService.java:119]: what about updating the supplier id?

File: src/main/java/com/mercuryCyclists/Inventory/service/ProductService.java [hunk 2]
```diff
@@ -93,15 +93,27 @@ public boolean addPart(Long productId, Part part) {
...
+    /**
+     * Update part helper method
+     * @param product
+     * @param part
+     * @return If given product contains given part return true else false
+     */
+    private boolean updatePartHelper(Product product, Part part) {
+        return product.getSet().contains(part);
+    }
+
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Inventory/service/ProductService.java:105]: My bad, I now get why this is for.
UpdatePartHelper is not the best naming but I'm happy with it

File: src/main/java/com/mercuryCyclists/Inventory/service/ProductService.java [hunk 3]
```diff
@@ -93,15 +93,27 @@ public boolean addPart(Long productId, Part part) {
         return true;
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Inventory/service/ProductService.java:93]: In your addPart method, you need to validate that the supplier exists. Same with the update

**Review by pedrooste — CHANGES_REQUESTED** [review]

File: src/main/java/com/mercuryCyclists/Inventory/controller/ProductController.java [hunk 4]
```diff
@@ -69,37 +73,66 @@ public ResponseEntity<String> deleteProduct(@PathVariable Long id) {
...
+     */
+    private boolean validSupplier(Long supplierId) {
+        try {
+            restfulService.getSupplier(supplierId).getStatusCodeValue();
+            return true;
+        } catch (HttpServerErrorException e) {
+            return false;
+        }
+    }
+
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Inventory/controller/ProductController.java:90]: should move and perform check in service

File: src/main/java/com/mercuryCyclists/Inventory/entity/Product.java [hunk 5]
```diff
@@ -13,8 +14,10 @@ public class Product {
...
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
-//    @Column(unique = true)
+    @Column(nullable = false)
     private String name;
+    @PositiveOrZero(message = "Price must be >= 0")
+    @Column(nullable = false)
     private Double price;
     private String comment;
     @OneToMany(targetEntity = Part.class, cascade = CascadeType.ALL)
```
Comment (Jack-Is-2048) [inline src/main/java/com/mercuryCyclists/Inventory/entity/Product.java:23]: Needs Long quantity member var

**Pull request closed without merging**